src/main/java/seedu/address/logic/parser/CalledCommandParser.java

```diff
@@ -13,21 +16,27 @@
  */
 public class CalledCommandParser implements Parser<CalledCommand> {
 
+    public static final String MESSAGE_INDEX_NOT_PARSED = "The index entered is not within the allowed range. "
+            + "The index for display cannot be bigger than 2147483647 (i.e., MAX_VALUE).";
+
     /**
      * Parses the given {@code String} of argument and returns a CalledCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public CalledCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args);
+        List<String> argList = argMultimap.getAllValues(EMPTY_PREFIX);
 
         Index index;
         try {
+            ParserUtil.tryParseIndex(argList);
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (IllegalValueException ive) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, CalledCommand.MESSAGE_USAGE), ive);
+        } catch (NumberFormatException e) {
+            throw new ParseException(String.format(MESSAGE_INDEX_NOT_PARSED), e);
```

Comment:
Come to think of it, issit possible to combine what you have done in tryParseIndex into the existing parseIndex (see line 34)? So there wouldnt be a need for 2 (similar) methods. 
Also, since the delete, called and done parsers all have this exact same try catch block, if you can encapsulate this whole thing inside the parseIndex/tryParseIndex, and get it to throw a parse exception, it would be a lot neater

Reply:
wait disregard the point abt combining the 2 methods. It might cause other places that use parseIndex to fail